server/homebrew.api.js

```diff
@@ -365,7 +374,7 @@ const api = {
 			// Compress brew text to binary before saving
 			brew.textBin = zlib.deflateRawSync(brew.text);
 			// Delete the non-binary text field since it's not needed anymore
-			brew.text = undefined;
+			if(!req?.keepText) brew.text = undefined;
```

Comment:
We should never want to keep the raw text. This defeats the purpose of the text compression.

Reply:
We only want it returned for the brews which have just been cleaned - but we don't need it kept in the saved data; only in the returned data. I'll change that function so that the data that is actually saved is not changed.

Reply:
This should now be resolved by the latest commit.

Reply:
Sorry, I'm still not clear why this change was needed. Moving the function definition of `cleanscript` to a different place shouldn't have impacted this.

Can you help me understand what broke when cleanScript was moved out of `lookup`?

Reply:
Not so much a thing that broke, but rather a change in process to remove an assumption.

Previously, the function would replace the tags in the brew, then set the number of detected tags to zero. However, it did this without actually running the function to check the text - and because when the brew is saved to Mongo, the text property is removed, running the check function again on the returned brew data would result in an error.

The current iteration adds a `returnText` property to the request, and when the save process detects the presence of this field, the text property is copied out of the data to be saved, and then added to the object to be returned post-save, so the data written to disk is unchanged, but the `text` property is returned to be checked again.

Reply:
Ok, so this is just to facilitate re-scanning multiple times?

Reply:
Essentially, yes - to allow the updated text to be checked to confirm that it has been cleaned correctly.

Reply:
I'm going to fiddle with this code for a bit, if that's ok.

Reply:
100%, please do.

Reply:
Just backed out the last two commits and made it automatically perform `lookup()` again after cleaning, so it just fetches the updated brew and scans it again.

`updateBrew()` is really only designed to return the brew stub, not the whole document, so I think adding the separate *keepText* flag might end up being an escape hatch into behavior it isn't designed to do. Instead, it might be worth a more direct refactor to `updateBrew()` if we decide we need to reconsider exactly what fields it returns, or if it should return the whole document.